types/underscore/index.d.ts

```diff
@@ -431,16 +431,22 @@ declare module _ {
         includes: UnderscoreStatic['contains'];
 
         /**
-        * Calls the method named by methodName on each value in the list. Any extra arguments passed to
-        * invoke will be forwarded on to the method invocation.
-        * @param list The element's in this list will each have the method `methodName` invoked.
-        * @param methodName The method's name to call on each element within `list`.
-        * @param arguments Additional arguments to pass to the method `methodName`.
-        **/
-        invoke<T extends {}>(
-            list: _.List<T>,
+         * Calls the method named by `methodName` on each value in
+         * `collection`. Any extra arguments passed to invoke will be forwarded
+         * on to the method invocation.
+         * @param collection The collection of elements to invoke `methodName`
+         * on.
+         * @param methodName The name of the method to call on each element in
+         * `collection`.
+         * @param args Additional arguments to pass to method `methodName`.
+         * @returns An array containing the result of the method call for each
+         * item in `collection`.
+         **/
+        invoke(
+            list: Collection<any>,
             methodName: string,
-            ...args: any[]): any;
+            ...args: any[]
+        ): any[];
```

Comment:
Is this the "magic" that requires TS >3?

```ts
invoke<T, K extends keyof T>(
    collection: Collection<T>,
    methodName: K,
    ...args: Parameters<T[K]>
): ReturnType<T[K]>;
```

Reply:
Yup, more or less. I do something a little more like `IterateeResult`, though, to handle `any` and unions and whatnot. I think it's all in a5e94d0405b0b280c37793df8949ba6383fe0f75, though it's possible that I might have made a minor tweak or two in later commits.

Reply:
Actually, though, in trying out some experiments it seems like in this case fancy stuff for any may not be necessary and fancy stuff for type unions may not be a good idea, so maybe I will just go with the simpler version here.